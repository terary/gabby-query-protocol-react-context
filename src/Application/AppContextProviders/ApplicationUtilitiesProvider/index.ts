import ApplicationUtilitiesProvider, {
  ApplicationUtilitiesContext,
  useTestUtilities,
} from "./ApplicationUtilitiesProvider";
import type { ApplicationUtilitiesType } from "./type";

export default ApplicationUtilitiesProvider;
export { ApplicationUtilitiesContext, useTestUtilities };
export type { ApplicationUtilitiesType };
