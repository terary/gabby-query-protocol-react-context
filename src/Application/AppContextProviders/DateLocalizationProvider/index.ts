import DateLocalizationProvider from "./Provider";

export default DateLocalizationProvider;
