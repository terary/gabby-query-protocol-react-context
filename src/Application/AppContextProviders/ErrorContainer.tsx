import { ReactNode } from "react";
import { ErrorBoundary } from "react-error-boundary";

function ErrorFallback({ error }: { error: Error }) {
  return (
    <div role="alert">
      <p>Something went wrong:</p>
      <pre>{error.message}</pre>
    </div>
  );
}

export const ErrorContainer = ({ children }: { children: ReactNode }) => {
  return <ErrorBoundary FallbackComponent={ErrorFallback}>{children}</ErrorBoundary>;
};
