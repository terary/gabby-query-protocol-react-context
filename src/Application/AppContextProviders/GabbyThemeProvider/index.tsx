import { GabbyThemeProvider } from "./Provider";
export default GabbyThemeProvider;
