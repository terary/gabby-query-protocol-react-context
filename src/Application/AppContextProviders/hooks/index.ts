import { useApplicationUtilities } from "./useApplicationUtilities";

export { useApplicationUtilities };
