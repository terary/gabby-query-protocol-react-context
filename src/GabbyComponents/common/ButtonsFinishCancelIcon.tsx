/* eslint-disable import/prefer-default-export */
import DoneSharpIcon from "@mui/icons-material/DoneSharp";
import ClearSharpIcon from "@mui/icons-material/ClearSharp";
import IconButton from "@mui/material/IconButton";
import Grid from "@mui/material/Grid";

import { useTranslation } from "react-i18next";

interface Props {
  onFinish: () => void;
  onCancel: () => void;
}

export const ButtonsFinishCancelIcon = ({ onFinish, onCancel }: Props): JSX.Element => {
  const { t, i18n } = useTranslation();
  return (
    <Grid container spacing="3">
      <Grid item xs={6}>
        <IconButton
          aria-label={t("ButtonsFinishCancelLarge_finish")}
          color="success"
          onClick={onFinish}
        >
          <DoneSharpIcon fontSize="medium" />
        </IconButton>
      </Grid>
      <Grid item xs={6}>
        <IconButton
          aria-label={t("ButtonsFinishCancelLarge_cancel")}
          color="error"
          onClick={onCancel}
        >
          <ClearSharpIcon fontSize="medium" />
        </IconButton>
      </Grid>
    </Grid>
  );
};
