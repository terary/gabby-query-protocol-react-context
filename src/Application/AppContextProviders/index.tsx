import React from "react";
import { TSupportedLocales } from "..";
import ApplicationUtilitiesProvider, {
  useTestUtilities,
} from "./ApplicationUtilitiesProvider";
import { ErrorContainer } from "./ErrorContainer";
import DateLocalizationProvider from "./DateLocalizationProvider";
import GabbyThemeProvider from "./GabbyThemeProvider";

interface IAppProviderParameters {
  children?: React.ReactNode;
  initialLocaleCode?: TSupportedLocales;
}

export const AppContextProviders = ({
  initialLocaleCode,
  children,
}: IAppProviderParameters) => {
  return (
    <ErrorContainer>
      {" "}
      {/* Error container may have issue with hooks, requires testing */}
      <ApplicationUtilitiesProvider initialLocaleCode={initialLocaleCode}>
        <GabbyThemeProvider>
          <DateLocalizationProvider>{children}</DateLocalizationProvider>
        </GabbyThemeProvider>
      </ApplicationUtilitiesProvider>
    </ErrorContainer>
  );
};

export { useTestUtilities };
