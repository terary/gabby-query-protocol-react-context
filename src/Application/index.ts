import { SUPPORTED_LOCALE_CODES_ARRAY, SUPPORTED_LOCALE_DICTIONARY } from "./i18n";
import { PageContentWrapper } from "./PageContentWrapper";
import { useApplicationUtilities } from "./AppContextProviders/hooks";
import { TSupportedLocales } from "./i18n";
import { customStyles } from "./custom-styles";
export {
  customStyles,
  PageContentWrapper,
  useApplicationUtilities,
  SUPPORTED_LOCALE_CODES_ARRAY,
  SUPPORTED_LOCALE_DICTIONARY,
};

export type { TSupportedLocales };
