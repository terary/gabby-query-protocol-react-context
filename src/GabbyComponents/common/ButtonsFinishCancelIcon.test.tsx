import React from "react";
import renderer from "react-test-renderer";
import { ButtonsFinishCancelIcon } from "./ButtonsFinishCancelIcon";
import { BrowserRouter } from "react-router-dom";
// import { AppContextProviders } from "../../Application/AppContextProviders";
import { AppContextProviders } from "../../Application/AppContextProviders";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <AppContextProviders>
        {/* <BrowserRouter> */}
        <ButtonsFinishCancelIcon onCancel={() => {}} onFinish={() => {}} />
        {/* </BrowserRouter> */}
      </AppContextProviders>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
