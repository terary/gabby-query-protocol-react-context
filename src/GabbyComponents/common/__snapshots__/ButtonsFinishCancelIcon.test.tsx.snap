// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  " ",
  <div
    className="Toastify"
  />,
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-uinpbw-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-ayy8wd-MuiGrid-root"
    >
      <button
        aria-label="Finalizado"
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSuccess MuiIconButton-sizeMedium css-1r6h6t4-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="DoneSharpIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M9 16.2 4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
          />
        </svg>
      </button>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-ayy8wd-MuiGrid-root"
    >
      <button
        aria-label="Cancelar"
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorError MuiIconButton-sizeMedium css-1jo1f9u-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ClearSharpIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z"
          />
        </svg>
      </button>
    </div>
  </div>,
]
`;
